### ISSUE TITLE
Error for "Send for Signature" in  Documentibm-content-navigator-samples/DocuSignSamplePlugin/DocuSignPlugin/

### ISSUE THREAD

**Issue by xavnicnsi** [ibm-ecm/ibm-content-navigator-samples#3]
Once integrated by following the readme guide, I got those errors in log file when clicking on "Send for Signature" from the docusign context menu: 
![image](https://user-images.githubusercontent.com/99875865/154965999-2deffd5d-4dfd-4467-966c-886a4bf84e66.png)

![image](https://user-images.githubusercontent.com/99875865/154965840-27d28f13-902a-400f-893c-32b0a919dd6f.png)

•	com.ibm.ecm.struts.actions.PluginAction.isSecureService() Invalid plugin id: DocuSignPlugin.  This is likely a logic error in the plugin itself.
•	com.ibm.ecm.struts.actions.BaseActionHandlerImpl.executeAction()
java.lang.NullPointerException

**Comment by emshedu** [comment]
Hi Please pull the latest update from Git. Please let me know if the issues still persists

**Comment by xavnicnsi** [comment]
Hello emshedu,

the previous error is gone and tahbks for that fix :) Really Appreciate your code. But I have now another issue. 
After filling up the form in icn, I enter 2 email addresses as recipient but after sending the Signature request I'm getting this error 
![image](https://user-images.githubusercontent.com/99875865/157715508-aa4f3f84-bca1-40ec-931f-74faff7dd674.png)

I checked the log file and I get this error: 
CIWEB Error: [XXXXXXXX@127.0.0.1] [REQUEST 1209] com.ibm.ecm.util.PluginUtil.invokeService()
java.lang.RuntimeException: Post request Failed - HTTP error code: 400 {  "errorCode": "INVALID_EMAIL_ADDRESS_FOR_RECIPIENT",  "message": "The email address for the recipient is invalid. The recipient Id follows."}
	at com.ibm.icn.extension.docusign.util.DocuSignUtil.executePostUrl(DocuSignUtil.java:167)

Same error stated in, Docusign Api dashboard
![image](https://user-images.githubusercontent.com/99875865/157716550-97aebfcb-e533-427e-b440-a7bc74ddd3b7.png)
any idea?

TIA

**Comment by emshedu** [comment]
Hi [xavnicnsi](https://github.com/xavnicnsi)
this is because the emails needs to be registered within Docusign API as signees

**Comment by xavnicnsi** [comment]
Wow my bad. I don't know why I thought that the email addresses didn't need to be registered in Docusign Account so each email address could be used even if not registered. But of course it makes sense in a way but in some other use cases, not sure...
Anyway once the email addresses are registered within the templates , then it's working. Many thanks again emshedu for your quick response. Really appreciate your code again!!

**Comment by emshedu** [comment]
You're very welcome! I am making some updates to add extra UI functionality in order to improve user experience. Please look forward to an updated version. (I will also inform this thread)
Thank you for utilising this plugin

**Comment by xavnicnsi** [comment]
Sure I'm looking forward to it.
By the way, once all recipient signed the document, is ICN supposed to get the new document with all recipients signatures? like a checkin process to replace the original doc sent without signature?

**Comment by xavnicnsi** [comment]
or the checkin process has to be done manually by recovering the signed document from the last email once docusign signing process has been completed?

**Comment by emshedu** [comment]
Yes you are correct! Please manually retrieve the signed document once it is complete. This is one of the updates that will be in the coming version. ICN will be able to retrieve the signed document, and update the tooltip related to the document with the the status of the plugin. This will be done periodically

**Comment by xavnicnsi** [comment]
That's great to hear. thanks for confirming that new feature! really appreciate your code!!

**Comment by wdrdres3qew5ts21** [comment]
Just adding more picture about how to config for "recipient email address" you need to create Template First, then put list of people who required to signed.
<img width="1686" alt="image" src="https://user-images.githubusercontent.com/25000903/233439471-aaa9ad3d-5adb-4473-86b7-c145b6bc8e39.png">
<img width="1698" alt="image" src="https://user-images.githubusercontent.com/25000903/233439570-1f284db1-1eb9-4909-bbfa-ec0792cd5071.png">
<img width="1395" alt="image" src="https://user-images.githubusercontent.com/25000903/233440111-c6cd177f-0ebe-4a1d-8356-c8ddfd4c2532.png">

**Comment by xavnicnsi** [comment]
Alternatively, a template without any recipient (not recommended) can be set to allow any recipient email adddress for the send for signature from ICN
